### DISCUSSION TITLE
Help with RegEx matching

### DISCUSSION THREAD

**Discussion by chiendinhfd** [HBOCodeLabs/wiremock-captain#396] [Q&A]
Hi all, using wiremock-captain to talk to a wiremock server on my local host.

Need some help with trying to do a path regex match.  I'm using the wiremock-captain plugin in js to talk to my wiremock servers and this is my iWireMockRequest 

```
const request: IWireMockRequest = {
	method: 'GET',
	endpoint: '\/contests\/\d+-\d+\/entries\?page=1&page_size=100&user=22',
}

const response: IWireMockResponse = {
	status: 200,
	body: ***,
}

 const features: IWireMockFeatures = {
	requestEndpointFeature: EndpointFeature.UrlPattern,
}
```
when I try to register the above to my wiremock server, i see in my logs that its registering the regex as:

`{"request":{"method":"GET","urlPattern":"/contests/d+-d+/entries?page=1&page_size=100&user=22"}
`

In which the /d+-d+/ looks incorrect and its not doing the regex correctly.  I've tried to use \\d+ instead of \d+ but instead of it showing up as \d+ when registered, it shows up as \\d+  so i'm at a loss as to the correct way to escape the regex

**Comment by satvik-s** [comment]
hi @chiendinhfd, i missed this message as i did not get tagged in it. if you are still looking for help here, feel free to re-start the issue and tag me
